### PR TITLE
Allow abstract methods to be used in JaversGetter

### PR DESCRIPTION
**Pull request by jonfreedman**
Regarding issue #548 these changes remove the restriction that methods cannot be abstract, this supports the situation where a concrete sub-class implements an abstract `@Id` annotated method, specifically for use with Google's AutoValue project.  In order to avoid breaking the existing TypeFactoryBeanIdTest JaversMember has a #looksLikeId property introduced which can be propagated from the sub-class to the super-class method to support the situation where a concrete sub-class implements a method adding the `@Id` annotation.

### CODE REVIEW

File: javers-core/src/test/groovy/org/javers/core/cases/AutoValueCase.groovy [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    diff.changes[0].left == 1
+    diff.changes[0].right == 2
+
+    first.author == 'Alice'
+    second.author == 'Bob'
+    second.changes.size() == 1
+    second.changes[0].propertyName == 'value'
+    second.changes[0].left == 1
+    second.changes[0].right == 2
+    second.snapshots[0].managedType.managedClass.looksLikeId.size() == looksLikeId
```
Comment (jonfreedman) [inline javers-core/src/test/groovy/org/javers/core/cases/AutoValueCase.groovy]: I'm sure there is a better way to test this?

File: javers-core/src/test/groovy/org/javers/core/cases/AutoValueCase.groovy [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    where:
+    label << ['basic javers', 'javers with bean mapping', 'javers with bean mapping and entity registered']
+    javers << [
+            JaversBuilder.javers().build(),
+            JaversBuilder.javers().withMappingStyle(MappingStyle.BEAN).build(),
+            JaversBuilder.javers().withMappingStyle(MappingStyle.BEAN).registerEntity(AbstractEntity.class).build()
+    ]
+    looksLikeId << [0, 1, 1]
```
Comment (jonfreedman) [inline javers-core/src/test/groovy/org/javers/core/cases/AutoValueCase.groovy]: The basic javers case can be removed, it's just here for illustration of why this whole change is necessary

**Comment by jonfreedman** [comment]
Please let me know if you need anything else on this PR or if it can be included in 3.3.1

**Comment by bartoszwalacik** [comment]
This code should be cleaned before merging, especially these multiply nested blocks in isOverriden(). Now i'm on devoxx, will back to you in a few days.

**Comment by jonfreedman** [comment]
Sure, enjoy the conference, I'm developing against a local snapshot for the moment.

File: javers-core/src/main/java/org/javers/common/reflection/JaversGetterFactory.java [hunk 3]
```diff
@@ -63,16 +69,35 @@ private static boolean isNotStatic(Method rawMethod) {
...
+
+    private static boolean isSubClass(final Method parent, final Method toCheck) {
+        return parent.getDeclaringClass().isAssignableFrom(toCheck.getDeclaringClass());
+    }
+
+    private static boolean sameMethodName(final Method parent, final Method toCheck) {
+        return parent.getName().equals(toCheck.getName());
+    }
+
+    private static boolean returnTypeCovariant(final Method parent, final Method toCheck) {
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/common/reflection/JaversGetterFactory.java]: +1 for taking care about return type covariance

**Pull request merged**